code/game/objects/structures/tank_holder.dm

```diff
@@ -0,0 +1,134 @@
+/obj/structure/tank_holder
+	name = "tank holder"
+	desc = "A metallic frame that can hold tanks and extinguishers."
+	icon = 'icons/obj/tank.dmi'
+	icon_state = "holder"
+
+	custom_materials = list(/datum/material/iron = 2000)
+
+	density = FALSE
+	anchored = FALSE
+	pass_flags = LETPASSTHROW
+	max_integrity = 20
+
+	resistance_flags = FIRE_PROOF
+
+	var/obj/item/tank = null
+
+/obj/structure/tank_holder/Initialize()
+	. = ..()
+	if(tank)
+		density = TRUE
+		tank = new tank(src)
+
+/obj/structure/tank_holder/Destroy()
+	if(tank)
+		tank.forceMove(get_turf(src))
+		tank = null
+	return ..()
+
+/obj/structure/tank_holder/CanAllowThrough(atom/movable/mover, turf/target)
+	. = ..()
+	if(istype(mover) && mover.throwing)
+		return TRUE
+
+/obj/structure/closet/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>It is held together by some <b>screws</b>.</span>"
+
+/obj/structure/tank_holder/attackby(obj/item/W, mob/user, params)
+	if(user.a_intent == INTENT_HARM)
+		return ..()
+	if(contents.len != 0)
+		to_chat(user, "<span class='warning'>There's already something in \the [src].</span>")
+		return
+
+	if(!W.tank_holder_icon_state)
+		to_chat(user, "<span class='warning'>\The [W] does not fit in \the [src].</span>")
+		return
+
+	if(!user.transferItemToLoc(W, src))
+		return
+	to_chat(user, "<span class='notice'>You put \the [W] into \the [src].</span>")
+	if(!user.transferItemToLoc(W, src))
+		return
+	after_put_tank(W)
+
+/obj/structure/tank_holder/screwdriver_act(mob/living/user, obj/item/I)
+	if(..())
+		return TRUE
+	if(tank)
+		return FALSE
+	I.play_tool_sound(src)
+	deconstruct(TRUE)
+	return TRUE
+
+/obj/structure/table_frame/deconstruct(disassembled = TRUE)
+	new /obj/item/stack/rods(get_turf(src), 2)
+	qdel(src)
+
+/obj/structure/tank_holder/attack_paw(mob/user)
+	return attack_hand(user)
+
+/obj/structure/tank_holder/attack_hand(mob/user)
+	if(!tank)
+		return ..()
+	to_chat(user, "<span class='notice'>You take \the [tank] from \the [src].</span>")
+	add_fingerprint(user)
+	user.put_in_hands(tank)
```

Comment:
need adjacency check first to prevent TK teleporting. 

if(Adjacent(user) && !issilicon(user))